torch/nn/utils/parametrizations.py

```diff
@@ -192,13 +467,11 @@ def spectral_norm(module: Module,
         >>> torch.linalg.matrix_norm(snm.weight, 2)
         tensor(1.0000, grad_fn=<CopyBackwards>)
     """
-    if not hasattr(module, name):
+    weight = getattr(module, name, None)
```

Comment:
Why do you need to change this?

Reply:
I changed it for consistency between the two parametrisations, as I believed it was better to avoid the two look ups (not that it changes much really). Do you think it is better the pattern `hasattr` + `getattr` or `getattr(..., None)`?

Reply:
I guess the getattr with None is more concise.